bin/scripts/continious-data-population.js

```diff
@@ -0,0 +1,1861 @@
+                defaultTemplates.forEach(function(item) {
+                    if (item._id === inputs[i].POPULATOR_TEMPLATE_ID) {
+                        template[inputs[i].APP_ID] = item;
+                    }
+                });
+                if (!template[inputs[i].APP_ID]) {
+                    db.close();
+                    writeMsg('INVALID POPULATOR_TEMPLATE_ID ', inputs[i].POPULATOR_TEMPLATE_ID);
+                }
+            }
+            if (!app[inputs[i].APP_ID]) {
+                db.close();
+                writeMsg('INVALID APP_ID ', inputs[i].APP_ID);
+            }
+            if (!template) {
+                db.close();
+                writeMsg('INVALID POPULATOR_TEMPLATE_ID ', inputs[i].POPULATOR_TEMPLATE_ID);
+            }
+            db.close();
+            users[inputs[i].APP_ID] = [];
+
+
+            pluginList[inputs[i].APP_ID] = await sendRequest({
+                requestType: 'GET',
+                Url: SERVER_URL + "/o/plugins?",
+                requestParamKeys: { app_id: inputs[i].APP_ID, api_key: API_KEY }
+            });
+        }
+    }
+    catch (error) {
+        clearInterval(interval);
+        writeMsg('There was an error while checking db parameters', error);
+    }
+}
+readDbParameters();
+
+var interval = setInterval(async function() {
+    for (var z = 0; z < Object.keys(inputs).length; z++) {
+        //users = [];
+        var userCount = getRandomInt(maxUserCount / 2, maxUserCount);
+        if (!users[inputs[z].APP_ID] || users[inputs[z].APP_ID].length === 0) {
+            for (var i = 0; i < userCount; i++) {
+                user = new getUser(template[inputs[z].APP_ID] && template[inputs[z].APP_ID].up);
+                users[inputs[z].APP_ID].push(user);
+            }
+        }
+        else {
+            if (users[inputs[z].APP_ID].length < maxUserCount) {
+                if (getRandomInt(1, 3) === 1) {
+                    user = new getUser(template && template.up);
+                    users[inputs[z].APP_ID].push(user);
+                }
+            }
+        }
+
+        var upperBoundary = Math.floor(userCount / 10);
+
+        var indexes = [];
+        for (var i0 = 0; i0 < upperBoundary; i0++) {
+            indexes.push(getRandomInt(0, users[inputs[z].APP_ID].length - 1));
+        }
+
+        indexes.forEach(function(index) {
+            if (users[inputs[z].APP_ID][index].hasSession) {
+                if (users[inputs[z].APP_ID][index].eventsInSession >= 1) {
+                    users[inputs[z].APP_ID][index].eventsInSession = 0;
+                    users[inputs[z].APP_ID][index].startSession(template[inputs[z].APP_ID], app[inputs[z].APP_ID], inputs[z].POPULATOR_TEMPLATE_ID);
+                }
+                else {
+                    users[inputs[z].APP_ID][index].addEvent(template[inputs[z].APP_ID], app[inputs[z].APP_ID], inputs[z].POPULATOR_TEMPLATE_ID);
+                    users[inputs[z].APP_ID][index].eventsInSession++;
```

Comment:
How many events do we add here?
Just one or more?

Reply:
it changes depending on the condition [here.](https://github.com/Countly/countly-server/pull/3454/files#diff-0c04b3f9535f8f004b58ca5b95264c0e117bbf0152040336f9151559ccd6d3f3R1074) 
If other events are concats to the Events object, more than 1 event can be added.